elements/lisk-transactions/src/base_transaction.ts

```diff
@@ -454,6 +456,52 @@ export abstract class BaseTransaction {
 		this._id = getId(this.getBytes());
 	}
 
+	/* tslint:disable:next-line: no-any no-null-keyword */
+	public dbRead(raw: any): TransactionJSON | null {
+		if (!raw.t_id) {
+			return null;
+		}
+
+		const transactionJSON: TransactionJSON & {
+			readonly requesterPublicKey: string;
+			readonly [key: string]: string | number | object | null;
+		} = {
+			id: raw.t_id,
+			height: raw.b_height,
+			blockId: raw.b_id || raw.t_blockId,
+			type: parseInt(raw.t_type, 10),
+			timestamp: parseInt(raw.t_timestamp, 10),
+			senderPublicKey: raw.t_senderPublicKey,
+			requesterPublicKey: raw.t_requesterPublicKey,
+			senderId: raw.t_senderId,
+			recipientId: raw.t_recipientId,
+			recipientPublicKey: raw.m_recipientPublicKey || null,
+			amount: raw.t_amount,
+			fee: raw.t_fee,
+			signature: raw.t_signature,
+			signSignature: raw.t_signSignature,
+			signatures: raw.t_signatures ? raw.t_signatures.split(',') : [],
+			confirmations: parseInt(raw.confirmations || 0, 10),
+			asset: {},
+		};
+
+		const transaction = Object.keys(
+			{ ...transactionJSON, asset: this.assetDbRead(raw) } || {},
+		).reduce(
+			(tx, key) => {
+				// Check for null or undefined
+				if (transactionJSON[key]) {
+					tx[key] = transactionJSON[key];
+				}
+
+				return tx;
+			},
+			{} as any,
```

Comment:
do we need `any` here? prefer to avoid `any` in source code

Reply:
This is to avoid tslint complaining about `no index signature`